_posts/2016-07-20-Automation-testing-with-postman.md

```diff
@@ -0,0 +1,166 @@
+<br/>
+<br/>
+Finally select the "Production" environment in the environment dropdown.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/manage_environment_production.png)
+
+### Creating a POST request
+First let's enter `{% raw %}{{url}}/WayneS/Calculator/0.1.0{% endraw %}` in the request field and change the method from GET to POST.  We also need some headers so add `Content-Type application/json`, `Authorization {% raw %}Simple {{key}}{% endraw %}` aswell.  Here we are using the environment variable `{% raw %}{{url}}{% endraw %}` and `{% raw %}{{key}}{% endraw %}` so that when we switch environments the variables are replaced accordingly.  The `{% raw %}{{...}}{% endraw %}` format can only be used in the request URL/URL params/Header values/form-data/url-encoded values/Raw body content/Helper fields.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_headers.png)
+<br/>
+<br/>
+Postman also has a few dynamic variables which you can use. For example `{% raw %}{{$guid}}{% endraw %}` a v4 style guid, `{% raw %}{{$timestamp}}{% endraw %}` the current timestamp, `{% raw %}{{$randomInt}}{% endraw %}` a random integer between 0 and 1000. More of those will be added in future releases. But for now let us just simply enter `"x=log(2)"` as the raw content of our request.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_body.png)
+<br/>
+<br/>
+Finally let's hit the "Send" button and if everything goes as expected we should receive the following response.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_send.jpg)
+<br/>
+<br/>
+Next we are going to write our test but first let us save our request into a collection. By clicking on the create collection button on the collections tab following modal will be presented.  Simply enter "Calculator" as the name of the collection and hit the create button.
+<br/>
+<br/>
+![Alt text]({{ '/img/postman-2016/create_collection.jpg'| prepend:site.baseurl }})
+<br/>
+<br/>
+Now hit the "Save" button next to the request field. Enter "Log" as the name of the request and select "Calculator" from the dropdown menu.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_save.jpg)
+
+### Writing a test
+A Postman test is essentially JavaScript code which sets values for the special 'tests' object. To know which other objects and libraries are available while writing your test cases make sure you check following [link](https://www.getpostman.com/docs/sandbox){:target="_blank"}. Let's copy following code snippet in the Tests sandbox.
+<br/>
+{% highlight bash %}tests["Status code is 200"] = responseCode.code === 200;
+var jsonData = JSON.parse(responseBody);
+tests["Verify result"] = jsonData.result.x === "0.69314718056";{% endhighlight %}
+<br/>
+![Alt text](/img/postman-2016/test.jpg)
+<br/>
+<br/>
+The test will run each time you hit the "Send" button. If you need for example an other functions for setting variables you can write them for example in the pre-request sandbox as shown below:
```

Comment:
an other functions for setting variables you can...
an other function for setting variables, you can write it
some other functions for setting variables, you can write them...